homeassistant/components/intelliclima/coordinator.py

```diff
@@ -0,0 +1,55 @@
+"""DataUpdateCoordinator for IntelliClima."""
+
+from datetime import timedelta
+
+from pyintelliclima import IntelliClimaAPI, IntelliClimaAPIError, IntelliClimaDevices
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, LOGGER
+
+type IntelliClimaConfigEntry = ConfigEntry[IntelliClimaCoordinator]
+
+
+class IntelliClimaCoordinator(DataUpdateCoordinator[IntelliClimaDevices]):
+    """Coordinator to manage fetching IntelliClima data."""
+
+    def __init__(
+        self, hass: HomeAssistant, entry: IntelliClimaConfigEntry, api: IntelliClimaAPI
+    ) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
+            config_entry=entry,
+        )
+        self.api = api
+        self.data = IntelliClimaDevices.empty()
+
+    async def _async_setup(self) -> None:
+        """Set up the coordinator - called once during first refresh."""
+        # Authenticate and get initial device list
+        try:
+            await self.api.authenticate()
+            self.data = await self._async_update_data()
+            LOGGER.info(
+                "Discovered %d IntelliClima VMC device(s)",
+                len(self.data.ecocomfort2),
+            )
```

Comment:
```suggestion
```
It is not needed. See: https://github.com/home-assistant/core/blob/b883d2f51967713420e3c8b7e3229d4ad8283df1/homeassistant/helpers/update_coordinator.py#L342-L349